File: src/getImage.js
a = [];
$(function() {
    var params = {
        // Request parameters
        "q": "code programming",
        "count": "1",
        "offset": "0",
        "mkt": "en-us",
        "safeSearch": "Strict",
    };
    
    $.ajax({
        url: "https://api.cognitive.microsoft.com/bing/v5.0/images/search?" + $.param(params),
        beforeSend: function(xhrObj){
            // Request headers
            xhrObj.setRequestHeader("Ocp-Apim-Subscription-Key", keys.microsoftapi);
        },
        type: "GET",
        // Request body
        data: "{body}",
    })
    .done(function(data) {
        a = data;
        $('#slide').attr("src", data['value'][0]['contentUrl'])
        alert("success");
    })
    .fail(function() {
        alert("error");
    });
});

File: src/GetImage.py
import http.client, urllib.request, urllib.parse, urllib.error, base64, json
from pprint import pprint

class GetImage:

    def __init__(self, key):
        self.key = key

    def getImage(self, keywords):

        search_string = ""
        for x in keywords:
            search_string = search_string + " " + x

        headers = {
            # Request headers
            'Ocp-Apim-Subscription-Key': self.key,
        }

        params = urllib.parse.urlencode({
            # Request parameters
            'q': search_string,
            'count': '1',
            'offset': '0',
            'mkt': 'en-us',
            'safeSearch': 'Strict',
        })

        try:
            conn = http.client.HTTPSConnection('api.cognitive.microsoft.com')
            conn.request("GET", "/bing/v5.0/images/search?%s" % params, "{body}", headers)
            response = conn.getresponse()
            data = json.loads(response.read().decode('utf-8'))
            conn.close()

            return data['value'][0]['contentUrl']
        except Exception as e:
            print("[Errno {0}] {1}".format(e.errno, e.strerror))

